### PR TITLE
Render title sequence to screenshots

### PR DESCRIPTION
**Pull request by janisozaur**
This is specifically for @spacek531 and @PFCKrutonium .

This will ensure proper advancing of frames and render all visible frames to PNG screenshots, so you can record title sequence _properly_.

Make sure you have "software" renderer and uncap FPS turned off. Screenshot will be of your window size, you can edit that in `config.ini` too.

### CODE REVIEW

**Comment by PFCKrutonium** [comment]
This is Specifically Awesome. How does it work?

**Comment by janisozaur** [comment]
Just make sure the options are all set properly and launch the headful client. It will dump all the screenshots at the usual location.

**Comment by PFCKrutonium** [comment]
Ah okay

**Comment by janisozaur** [comment]
you should also disable displaying FPS counter, it will show the render speed, not the playback speed. Remember also that this is 40 FPS

**Comment by janisozaur** [comment]
Closing, as this is not intended to be merged, at least not in such form. I still had to make a PR so that CIs would build it.

**Comment by spacek531** [comment]
When/if this gets upgraded to interpolate frames, can FFmpeg be looked into as a video packaging utility? It can accept pipes which my research tells me would allow the program to skip writing every frame to the hard drive and later removing them from the hard drive. With the right options, the music could be overlaid automatically.

**Comment by janisozaur** [comment]
I rebased it to take some of the recent changes into account. This is still locked to `uncap_fps = false`.

**Comment by spacek531** [comment]
Is this for fixes to the draw order bugs that create the flashing in The Bayou and the tracks-appear-above-the-tram in Yokum Creek?

**Comment by janisozaur** [comment]
This PR is not fixing anything, although it may make the issue more easy to spot.

Here are some screenshots I produced that show flicker of the sign I mentioned:

screenshot425
screenshot450

File: src/openrct2/platform/shared.c [hunk 1]
```diff
@@ -146,6 +146,13 @@ void platform_refresh_video(bool recreate_window)
     gfx_invalidate_screen();
 }
 
+static uint32 _ticks = 0;
+
+void platform_advance_ticks()
+{
+    _ticks += 25;
```
Comment (janisozaur) [inline src/openrct2/platform/shared.c]: ms to advance ticks by, integer.

25 => 1000/25 = 40fps
16.67 => 1000/16.67 = 60fps
16 => 1000/16 = 62.5fps
17 => 1000/17 = 58.82fps
8 => 1000/8 = 125fps

**Comment by janisozaur** [comment]
@Krutonium this should now be able to render properly with `uncap_fps = true`. I marked the place you need to modify to set the frame rate, there's too much effort to make it a runtime option.

There are some odd things going on if you set it to 1, so don't do that.

The caveat is I only allow setting an integer there, so you won't get perfect 60fps. I'd recommend you use 8ms and have it render 125fps, that way encoder should be able to get a frame close enough to where it needs one.

That said, while thinking about it, I should be able to add an option to allow floats in there, which will be the ideal solution for exact 60fps.

**Comment by janisozaur** [comment]
The previous comment no longer holds, this has been converted to use real values, so you can get 60 FPS.

**Comment by IntelOrca** [comment]
> Closing, as this is not intended to be merged, at least not in such form. I still had to make a PR so that CIs would build it.

**Comment by janisozaur** [comment]
A showcase for this: https://youtu.be/BjiZYu-Xv4w

**Comment by ZxBiohazardZx** [comment]
still flickers on some tiles but i guess that is expected? note the station building flickering in that example

**Pull request closed without merging**